src/main/java/main/Ui.java

```diff
@@ -0,0 +1,195 @@
+            "/ /_\\\\ (_) | (_) | (_| | |_) | |_| |  __/\\_/ \n" +
+            "\\____/\\___/ \\___/ \\__,_|_.__/ \\__, |\\___\\/   \n" +
+            "                              |___/          ";
+
+    /**
+     * Displays the greeting message and logo to the user.
+     */
+    public static void displayGreeting() {
+        System.out.println("Hello! I'm\n" + LOGO + "\nWhat can I do for you?");
+        Ui.insertLine();
+    }
+
+    /**
+     * Displays the goodbye message to the user.
+     */
+    public static void displayGoodbye() {
+        Ui.insertLine();
+        System.out.println(GOODBYE);
+        Ui.insertLine();
+    }
+
+    /**
+     * Prints a horizontal line for visual separation in the UI.
+     */
+    public static void insertLine() {
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    /**
+     * Shows an error message to the user.
+     *
+     * @param message The error message to be displayed.
+     */
+    public static void showError(String message) {
+        Ui.insertLine();
+        System.out.println("ERROR: " + message);
+        Ui.insertLine();
+    }
+
+    /**
+     * Shows a date-time parse error message to the user.
+     *
+     * @param message The error message to be displayed.
+     */
+    public static void showDateTimeParseError(String message) {
+        Ui.insertLine();
+        System.out.println("Date format is wrong, please input dd-mm-yyyy hhmm: " + message);
+        Ui.insertLine();
+    }
+
+    /**
+     * Shows an unexpected error message to the user.
+     *
+     * @param message The error message to be displayed.
+     */
+    public static void showUnexpectedError(String message) {
+        Ui.insertLine();
+        System.out.println("UNEXPECTED ERROR: " + message);
+        Ui.insertLine();
+    }
+
+    /**
+     * Displays the list of tasks to the user.
+     *
+     * @param tasks The list of tasks to be displayed.
+     */
+    public static void showTaskList(TaskList tasks) {
+        int length = tasks.size();
+        Ui.insertLine();
+        for (int i = 0; i < length; i++) {
+            Task t = tasks.get(i);
+            int num = i + 1;
+            System.out.println(num + "." + t);
+        }
+        Ui.insertLine();
+    }
+
+    /**
+     * Prints a message confirming that a deadline task has been added.
+     *
+     * @param tasks The list of tasks after the addition.
+     */
+    public static void printAddDeadline(TaskList tasks) {
+        int num = tasks.size();
+        Ui.insertLine();
+        System.out.println("Got it. I've added this task:\n" + tasks.get(num - 1));
+        System.out.println("Now you have " + num + " tasks in the list.");
+        Ui.insertLine();
+    }
+
+    /**
+     * Prints a message confirming that an event task has been added.
+     *
+     * @param tasks The list of tasks after the addition.
+     */
+    public static void printEvent(TaskList tasks) {
+        int num = tasks.size();
+        Ui.insertLine();
```

Comment:
Do you need another method for the insert line format? Since Ui.insertLine() is added at the start and end of the methods in this class.